src/env.h

```diff
@@ -726,6 +731,8 @@ class AsyncHooks : public MemoryRetainer {
   AliasedFloat64Array async_id_fields_;
 
   void grow_async_ids_stack();
+
+  v8::Persistent<v8::Value> execution_async_resource_;
```

Comment:
This needs to be switched to a stack too, but generally it’s nicer to use `v8::Global` because that automatically resets + provides move operators (which you’d probably need for turning this into a stack) – adding a lint rule in https://github.com/nodejs/node/pull/31018

Reply:
Yep, working on it but it's not cooperating so far. :sweat_smile: 

Anyway, thanks for the note about `v8::Global`. I'll give that a try!

Reply:
Switched to a stack of `v8::Global` values now. :smile_cat: